project_stock/models/project_task.py

```diff
@@ -0,0 +1,249 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class Task(models.Model):
+    _inherit = "project.task"
+
+    scrap_ids = fields.One2many(
+        comodel_name="stock.scrap", inverse_name="task_id", string="Scraps"
+    )
+    scrap_count = fields.Integer(
+        compute="_compute_scrap_move_count", string="Scrap Move"
+    )
+    move_ids = fields.One2many(
+        comodel_name="stock.move",
+        inverse_name="raw_material_task_id",
+        string="Stock Moves",
+        copy=False,
+        domain=[("scrapped", "=", False)],
+    )
+    use_stock_moves = fields.Boolean(related="stage_id.use_stock_moves")
+    done_stock_moves = fields.Boolean(related="stage_id.done_stock_moves")
+    stock_moves_is_locked = fields.Boolean(default=True)
+    allow_moves_action_confirm = fields.Boolean(
+        compute="_compute_allow_moves_action_confirm"
+    )
+    allow_moves_action_assign = fields.Boolean(
+        compute="_compute_allow_moves_action_assign"
+    )
+    stock_state = fields.Selection(
+        selection=[
+            ("pending", "Pending"),
+            ("confirmed", "Confirmed"),
+            ("assigned", "Assigned"),
+            ("done", "Done"),
+            ("cancel", "Cancel"),
+        ],
+        compute="_compute_stock_state",
+    )
+    picking_type_id = fields.Many2one(
+        comodel_name="stock.picking.type",
+        string="Operation Type",
+        readonly=False,
+        domain="[('company_id', '=', company_id)]",
```

Comment:
According to https://www.odoo.com/documentation/13.0/developer/howtos/company.html#multi-company-consistency it's not necessary.